projects/ngrx.io/content/guide/component-store/usage.md

```diff
@@ -0,0 +1,240 @@
+
+<div class="alert is-helpful">
+
+You can see the examples at Stackblitz:
+* "PaginatorComponent providing ComponentStore" <live-example name="component-store-paginator" noDownload></live-example>
+* "PaginatorComponent with PaginatorStore Service" <live-example name="component-store-paginator-service" noDownload></live-example>
+
+</div>
+
+<code-tabs linenums="true">
+  <code-pane
+    header="PaginatorComponent with PaginatorStore Service"
+    path="component-store-paginator-service/src/app/paginator.component.ts">
+  </code-pane>
+  <code-pane
+    header="PaginatorComponent providing ComponentStore"
+    path="component-store-paginator/src/app/paginator.component.ts">
+  </code-pane>
+  <code-pane
+    header="paginator.store.ts"
+    path="component-store-paginator-service/src/app/paginator.store.ts">
+  </code-pane>
+</code-tabs>
+
+#### Updating the state
+
+With `ComponentStore` extracted into `PaginatorStore`, the developer is now using updater and effects to update the state. Effects are used to perform additional validation and get extra information from sources with derived data (e.i. selectors).
+
+<code-example
+  path="component-store-paginator-service/src/app/paginator.component.ts"
+  region="updating-state"></code-example>
+
+#### Reading the state
+
+`PaginatorStore` exposes the two properties: `vm$` for aggregated *ViewModel* to be used in the template, and `page$` that would emitted whenever data aggregated of `PageEvent` changes.
+
+<code-tabs>
+  <code-pane
+    header="paginator.component.ts"
+    path="component-store-paginator-service/src/app/paginator.component.ts"
+    region="selectors"
+    >
+  </code-pane>
+  <code-pane
+    header="paginator.store.ts"
+    path="component-store-paginator-service/src/app/paginator.store.ts"
+    region="selectors">
+  </code-pane>
+</code-tabs>
+
+<div class="alert is-helpful">
+
+`page$` would emit `PageEvent` when page size or page index changes, however in some cases updater would update both of these properties at the same time. To avoid "intermediary" emissions, `page$` selector is using **`{debounce: true}`** configuration. This schedules a microtask, which allows the state to "settle", meaning all the synchronous changes to happen, and then it would emit a value. All of this happens before Angular's Change Detection runs. More about debouncing can be found in [selector section](guide/component-store/read#debounce-selectors).
+
+</div>
+
+### Local UI State patterns
+
+Components that use `ComponentStore` for managing **Local UI State** are frequently calling `updater`s directly.
+
+Effects can be also called when:
+* side effects are required (e.g. `event.stopPropagation()`)
+* derived data (from selectors) is needed to influence the new state
+* they are orchestrating a number of well-defined updaters
+
+The last point sometimes could be refactored into `updater`. Use your best judgement.
```

Comment:
Not exactly sure what you mean by refactoring orchestration of multiple updaters into a single updater.

Suggestion 1
```suggestion
The last point can sometimes be refactored into an aggregate `updater`. Use your best judgement.
```

Suggestion 2
```suggestion
The last point can sometimes be refactored into a single `updater`. Use your best judgement.
```

Suggestion 3
```suggestion
The last point can sometimes be refactored into another `updater`. Use your best judgement.
```

Reply:
The 3-rd option is what I meant :)